src/v/cluster/config_manager.cc

```diff
@@ -394,6 +394,16 @@ config_manager::preload_join(const controller_join_snapshot& snap) {
 
 ss::future<config_manager::preload_result>
 config_manager::preload(YAML::Node const& legacy_config) {
+    /*
+     * Typically, predefined config values are set in the property constructors;
+     * however, constraints rely on the existence of other config properties
+     * which are constructed when the config store is created. Therefore,
+     * predefined constraints are set here instead of in property constructors.
+     */
+    co_await ss::smp::invoke_on_all([] {
+        config::shard_local_cfg().constraints(config::predefined_constraints());
+    });
+
```

Comment:
The storage changes will be in a separate PR following https://github.com/redpanda-data/redpanda/issues/14018.
So I may move this change to there.